modules/rostests/apitests/zipfldr/zipfldr.cpp

```diff
@@ -0,0 +1,536 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Test for zipfldr
+ * COPYRIGHT:   Copyright 2020 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#include "precomp.h"
+
+#ifndef FILE_ATTRIBUTE_VIRTUAL
+#define FILE_ATTRIBUTE_VIRTUAL 0x10000
+#endif
+
+DEFINE_GUID(CLSID_ZipFolderContextMenu,    0xb8cdcb65, 0xb1bf, 0x4b42, 0x94, 0x28, 0x1d, 0xfd, 0xb7, 0xee, 0x92, 0xaf);
+
+
+static bool g_bOldZipfldr;
+const char test_file_1_contents[] = "Some generic text in the root file.\r\nMore text on a new line.";
+const char test_file_2_contents[] = "Some generic text in the file in folder_1.\r\nMore text on a new line.";
+
+static BOOL write_raw_file(const WCHAR* FileName, const void* Data, DWORD Size)
+{
+    BOOL Success;
+    DWORD dwWritten;
+    HANDLE Handle = CreateFileW(FileName, GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+
+    if (Handle == INVALID_HANDLE_VALUE)
+    {
+        skip("Failed to create temp file %ls, error %lu\n", FileName, GetLastError());
+        return FALSE;
+    }
+    Success = WriteFile(Handle, Data, Size, &dwWritten, NULL);
+    ok(Success == TRUE, "WriteFile failed with %lu\n", GetLastError());
+    ok(dwWritten == Size, "WriteFile wrote %lu bytes instead of %lu\n", dwWritten, Size);
+    CloseHandle(Handle);
+    return Success && (dwWritten == Size);
+}
+
+BOOL extract_resource(WCHAR* Filename, LPCWSTR ResourceName)
+{
+    WCHAR workdir[MAX_PATH];
+    GetTempPathW(_countof(workdir), workdir);
+    StringCchPrintfW(Filename, MAX_PATH, L"%sTMP%u.zip", workdir, GetTickCount() & 0xffff);
+
+    HMODULE hMod = GetModuleHandleW(NULL);
+    HRSRC hRsrc = FindResourceW(hMod, ResourceName, MAKEINTRESOURCEW(RT_RCDATA));
+    ok(!!hRsrc, "Unable to find %s\n", wine_dbgstr_w(ResourceName));
+    if (!hRsrc)
```

Comment:
add skip

Reply:
There is already an error indicating why the test aborted.